components/Pagination/Pagination.js

```diff
@@ -123,25 +141,36 @@ class Pagination extends React.Component {
         </li>
       );
     }
-
     return (
       <div className={`${cssClass}  content-view-pf-pagination`}>
-        <span>
-          <span className="pagination-pf-items-current">
-            {(currentPage + 1) * pageSize - pageSize + 1}
-            <span> - </span>
-            {currentPage === totalPages && totalItems || (currentPage + 1) * pageSize}
-          </span> of {totalItems}
+        <FormattedMessage
+          defaultMessage="{items} {start} {to} {end} of {total} {items}"
+          values={{
+            items: <span className="sr-only"><FormattedMessage defaultMessage="Items" /></span>,
+            start: startItems,
+            to: <span className="pagination-cmpsr-to"><FormattedMessage defaultMessage="to" /></span>,
```

Comment:
This is the first time I see `<FormattedMessage>` and reactl-intl (what a pain, compared to the usual gettext!), so excuse me thinking aloud: Why does "to" get translated separately, but "of" doesn't? I thought the outer `"{items} {start} {to} {end} of {total} {items}"` is the string that would actually end up being translated, so both the words themselves and their order can change.

I figure the difference is that the "to" gets wrapped into this extra `pagination-cmpsr-to` span, and you want to avoid putting that part into the translatable string. This makes sense, and the actual "to" word can still be translated (although it might be a little confusing to do that without seeing the context).

The same situation arises with normal gettext, and there's usually no better solution than "avoid formatting within one sentence".

So bottom line, this looks okay to me.

Reply:
I had pushed an updated version of this yesterday. Originally with this implementation, I was hoping to align with the PF design by using a dash between start and end values, like "1-50", but provide a friendlier version to screen reader users by using "to", as in "one to fifty". But there's no clean way to provide this in a way that sends the string to translators as "{startNumber} to {endNumber} of {totalNumber}". 

In my commit yesterday, I simplified the strings so that translators get "{startNumber} - {endNumber} of {totalNumber}", and screen readers get something like "(Current Items:) one dash fifty of one hundred". I still don't like the dash (-)—I'm not sure how well a dash works for either translation or screen reader users—but I also would like to be consistent with PF designs.

I'm still on the fence for this one, so I could easily be convinced one way or the other :-)